core/components/molecules/dialog/dialog.js

```diff
@@ -34,75 +35,165 @@ const createButtonForAction = (action, index) => {
   )
 }
 
-const Dialog = props => (
-  <Overlay {...props}>
-    <DialogElement width={props.width} {...Automation('dialog')}>
-      <DialogTitleBar {...Automation('dialog.title')}>
-        <span>{props.title}</span>
-        <Link onClick={props.onClose}>
-          <Icon name="close" size={16} />
-        </Link>
-      </DialogTitleBar>
-      <DialogBody {...Automation('dialog.body')}>{props.children}</DialogBody>
-      <DialogFooter {...Automation('dialog.footer')}>
-        <ButtonGroup>{props.actions.map(createButtonForAction)}</ButtonGroup>
-      </DialogFooter>
-    </DialogElement>
-  </Overlay>
-)
-
-const DialogElement = styled.div`
+const focusOnFormInput = ({ current }) => {
+  const node = ReactDOM.findDOMNode(current)
+  const form = node.querySelector('form')
+  if (!form) return
+
+  const firstInput = form.querySelector('input')
+  if (!firstInput) return
+
+  firstInput.focus()
+}
+
+const roleDependantProp = (props, requiredRole, propObject) =>
```

Comment:
Sorry, what does this do?

I'm also an accessibility noob 😅

Reply:
Basically sets a prop depending on the dialog's role (See auth0/cosmos/pull/1102/files/4ea9b610e8a523e003c6c7f757289d30777eb4fe#diff-dedf8280fcbfc0d57cb9fbfb4df7f85cR72)

Reply:
Ah got it, can we call the function `getAccessibilityRole`?

Reply:
If you want.. That would not be a problem for me